### PR TITLE
A user can deploy a process definition with a script task

### LINKED ISSUE #7155: A user can deploy a process definition with a script task
**Is your feature request related to a problem? Please describe.**

As a user I can deploy a process definition with a script task. The script task will behave like a service task, and the main purpose is to support the modelling of script task.

**Describe the solution you'd like**

A script task can be deployed and supports all service task properties, i.e. type, retries, input/output mappings and task headers

The `BPMN_ELEMENT_TYPE` will be `SCRIPT_TASK`.

**Describe alternatives you've considered**

- Allow users to execute scripts inside the workflow engine, i.e. using FEEL expressions (#4213)

**Additional context**

```
<bpmn:scriptTask id="Task" name="Task">
  <bpmn:extensionElements>
    <zeebe:taskDefinition type="task-type" retries="123" />
    <zeebe:ioMapping>
      <zeebe:input source="= processVariable" target="inputVariable" />
      <zeebe:output source="= localVariable" target="outputVariable" />
    </zeebe:ioMapping>
    <zeebe:taskHeaders>
      <zeebe:header key="firstTaskHeader" value="first task header value" />
      <zeebe:header key="secondTaskHeader" value="second task header value" />
    </zeebe:taskHeaders>
  </bpmn:extensionElements>
  <bpmn:incoming>IncomingFlow</bpmn:incoming>
  <bpmn:outgoing>OutgoingFlow</bpmn:outgoing>
</bpmn:scriptTask>
```

### PR DESCRIPTION
**Pull request by saig0**
## Description

* allow processes with script tasks
* script tasks behave exactly like service tasks

## Related issues

closes #7155 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [x] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [x] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by saig0** [comment]
I tested the behavior manually.

image

File: engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/transformer/FlowElementInstantiationTransformer.java [hunk 1]
```diff
@@ -62,12 +63,13 @@
...
     ELEMENT_FACTORIES.put(ParallelGateway.class, ExecutableFlowNode::new);
+    ELEMENT_FACTORIES.put(ReceiveTask.class, ExecutableReceiveTask::new);
+    ELEMENT_FACTORIES.put(ScriptTask.class, ExecutableJobWorkerTask::new);
     ELEMENT_FACTORIES.put(SequenceFlow.class, ExecutableSequenceFlow::new);
     ELEMENT_FACTORIES.put(ServiceTask.class, ExecutableJobWorkerTask::new);
-    ELEMENT_FACTORIES.put(UserTask.class, ExecutableJobWorkerTask::new);
-    ELEMENT_FACTORIES.put(ReceiveTask.class, ExecutableReceiveTask::new);
     ELEMENT_FACTORIES.put(StartEvent.class, ExecutableStartEvent::new);
     ELEMENT_FACTORIES.put(SubProcess.class, ExecutableFlowElementContainer::new);
+    ELEMENT_FACTORIES.put(UserTask.class, ExecutableJobWorkerTask::new);
```
Comment (korthout) [inline engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/transformer/FlowElementInstantiationTransformer.java:72]: abc love it 😍

File: bpmn-model/src/test/java/io/camunda/zeebe/model/bpmn/validation/ExpectedValidationResult.java [hunk 2]
```diff
@@ -64,7 +64,7 @@ public boolean matches(final ValidationResult result) {
       match = false;
     }
 
-    if (!expectedMessage.equals(result.getMessage())) {
+    if (!result.getMessage().startsWith(expectedMessage)) {
```
Comment (korthout) [inline bpmn-model/src/test/java/io/camunda/zeebe/model/bpmn/validation/ExpectedValidationResult.java:67]: Why was this necessary?

File: bpmn-model/src/test/java/io/camunda/zeebe/model/bpmn/validation/ZeebeJobWorkerTaskValidationTest.java [hunk 3]
```diff
@@ -16,153 +16,126 @@
...
+    taskModifier.accept(jobWorkerTaskBuilder);
+    return jobWorkerTaskBuilder.endEvent().done();
+  }
+
+  private static Stream<JobWorkerTaskBuilder> jobWorkerTaskBuilderProvider() {
+    return Stream.of(
+        JobWorkerTaskBuilder.of("serviceTask", AbstractFlowNodeBuilder::serviceTask),
+        JobWorkerTaskBuilder.of("businessRuleTask", AbstractFlowNodeBuilder::businessRuleTask),
+        JobWorkerTaskBuilder.of("scriptTask", AbstractFlowNodeBuilder::scriptTask));
+  }
```
Comment (korthout) [inline bpmn-model/src/test/java/io/camunda/zeebe/model/bpmn/validation/ZeebeJobWorkerTaskValidationTest.java:110]: Can this use the more general: `io.camunda.zeebe.engine.util.JobWorkerTaskBuilderProvider`?

**Review by korthout — APPROVED** [review]
LGTM 👍 but please have a look at my comments

File: bpmn-model/src/test/java/io/camunda/zeebe/model/bpmn/validation/ExpectedValidationResult.java (hunk 2 above)
Comment (saig0) [reply bpmn-model/src/test/java/io/camunda/zeebe/model/bpmn/validation/ExpectedValidationResult.java:67]: It was not necessary. But I aligned the class with the one from the engine module. 
We have this class in both models :sweat_smile:

Comment (korthout) [reply bpmn-model/src/test/java/io/camunda/zeebe/model/bpmn/validation/ExpectedValidationResult.java:67]: ok thanks for explaining

File: bpmn-model/src/test/java/io/camunda/zeebe/model/bpmn/validation/ZeebeJobWorkerTaskValidationTest.java (hunk 3 above)
Comment (saig0) [reply bpmn-model/src/test/java/io/camunda/zeebe/model/bpmn/validation/ZeebeJobWorkerTaskValidationTest.java:110]: No. Here, we're in the model API module. The other provider is in the engine module :sweat_smile:

Comment (korthout) [reply bpmn-model/src/test/java/io/camunda/zeebe/model/bpmn/validation/ZeebeJobWorkerTaskValidationTest.java:110]: Aah, that makes sense as well. Sorry for the stupid question 😄
